### PR TITLE
itsycal: init at 0.15.3

### PR DESCRIPTION
**Pull request by 4evy**
## Description of changes

A small macOS calendar program

GitHub Repo: https://github.com/sfsam/Itsycal

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/darwin/itsycal/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+, stdenvNoCC
+, unzip
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "itsycal";
+  version = "0.15.3";
+
+  src = fetchurl {
+    url = "https://web.archive.org/web/20240326164201/https://itsycal.s3.amazonaws.com/Itsycal.zip";
```
Comment (stepbrobd) [inline pkgs/os-specific/darwin/itsycal/default.nix:12]: They have versioned download links:

`https://itsycal.s3.amazonaws.com/Itsycal-<MAJOR.MINOR.PATCH>.zip`

Source:
[1] https://formulae.brew.sh/cask/itsycal#default
[2] https://formulae.brew.sh/api/cask/itsycal.json

File: pkgs/os-specific/darwin/itsycal/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    mkdir -p $out/Applications
+    mv Itsycal.app $out/Applications
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Itsycal is a tiny menu bar calendar";
+    homepage = "https://www.mowglii.com/itsycal/";
+    license = lib.licenses.free;
```
Comment (stepbrobd) [inline pkgs/os-specific/darwin/itsycal/default.nix:32]: I can't find any explicit licensing statement on their website, probably we should use `unfree` here?

**Review by stepbrobd — COMMENTED** [review]
Looking good!

I'm not sure if you are aware of #293498, the packaging process should be way easier after merging. I suggest keep track with #293498 and change the current `stdenvNoCC.mkDerivation` approach to `darwin.installBinaryPackage` after it merges.

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 299257` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by DontEatOreo** [comment]
> Looking good!
> 
> I'm not sure if you are aware of #293498, the packaging process should be way easier after merging. I suggest keep track with #293498 and change the current `stdenvNoCC.mkDerivation` approach to `darwin.installBinaryPackage` after it merges.

Thanks for notifying me about this I wasn't aware! I will keep track.

File: pkgs/os-specific/darwin/itsycal/default.nix (hunk 2 above)
Comment (DontEatOreo) [reply pkgs/os-specific/darwin/itsycal/default.nix:32]: I think you're looking at the old commit as it's `free` on commit 52269f864923947582e230028dbec096cafdee78 although I will change to mit

Comment (stepbrobd) [reply pkgs/os-specific/darwin/itsycal/default.nix:32]: You are right! It's MIT: https://github.com/sfsam/Itsycal/blob/master/LICENSE.txt

File: pkgs/os-specific/darwin/itsycal/default.nix (hunk 1 above)
Comment (DontEatOreo) [reply pkgs/os-specific/darwin/itsycal/default.nix]: Fixed

**Pull request closed without merging**